syntax = "proto3";

package taraxa.vm.statedb;

option go_package = "grpc_go";

import "google/protobuf/empty.proto";
import "common.proto";

service StateDB {
    rpc Put (BytesMessage) returns (google.protobuf.Empty);
    rpc Delete (BytesMessage) returns (google.protobuf.Empty);
    rpc Get (BytesMessage) returns (BytesMessage);
    rpc Has (BytesMessage) returns (BoolMessage);
    rpc Close (VmId) returns (google.protobuf.Empty);
}
